model/atmosphere/diffusion/tests/diffusion/integration_tests/test_benchmark_diffusion.py

```diff
@@ -0,0 +1,220 @@
+
+    diffusion_parameters = diffusion.DiffusionParams(config)
+
+    grid_manager = grid_utils.get_grid_manager_from_identifier(
+        grid, num_levels=10, keep_skip_values=True, backend=backend
+    )
+
+    mesh = grid_manager.grid
+    coordinates = grid_manager.coordinates
+    geometry_input_fields = grid_manager.geometry_fields
+
+    decomposition_info = construct_decomposition_info(mesh, backend)
+
+    geometry_field_source = grid_geometry.GridGeometry(
+        grid=mesh,
+        decomposition_info=decomposition_info,
+        backend=backend,
+        coordinates=coordinates,
+        extra_fields=geometry_input_fields,
+        metadata=geometry_meta.attrs,
+    )
+
+    cell_geometry = grid_states.CellParams(
+        cell_center_lat=geometry_field_source.get(geometry_meta.CELL_LAT),
+        cell_center_lon=geometry_field_source.get(geometry_meta.CELL_LON),
+        area=geometry_field_source.get(geometry_meta.CELL_AREA),
+    )
+    edge_geometry = grid_states.EdgeParams(
+        edge_center_lat=geometry_field_source.get(geometry_meta.EDGE_LAT),
+        edge_center_lon=geometry_field_source.get(geometry_meta.EDGE_LON),
+        tangent_orientation=geometry_field_source.get(geometry_meta.TANGENT_ORIENTATION),
+        coriolis_frequency=geometry_field_source.get(geometry_meta.CORIOLIS_PARAMETER),
+        edge_areas=geometry_field_source.get(geometry_meta.EDGE_AREA),
+        primal_edge_lengths=geometry_field_source.get(geometry_meta.EDGE_LENGTH),
+        inverse_primal_edge_lengths=geometry_field_source.get(
+            f"inverse_of_{geometry_meta.EDGE_LENGTH}"
+        ),
+        dual_edge_lengths=geometry_field_source.get(geometry_meta.DUAL_EDGE_LENGTH),
+        inverse_dual_edge_lengths=geometry_field_source.get(
+            f"inverse_of_{geometry_meta.DUAL_EDGE_LENGTH}"
+        ),
+        inverse_vertex_vertex_lengths=geometry_field_source.get(
+            f"inverse_of_{geometry_meta.VERTEX_VERTEX_LENGTH}"
+        ),
+        primal_normal_x=geometry_field_source.get(geometry_meta.EDGE_NORMAL_U),
+        primal_normal_y=geometry_field_source.get(geometry_meta.EDGE_NORMAL_V),
+        primal_normal_cell_x=geometry_field_source.get(geometry_meta.EDGE_NORMAL_CELL_U),
+        primal_normal_cell_y=geometry_field_source.get(geometry_meta.EDGE_NORMAL_CELL_V),
+        primal_normal_vert_x=geometry_field_source.get(geometry_meta.EDGE_NORMAL_VERTEX_U),
+        primal_normal_vert_y=geometry_field_source.get(geometry_meta.EDGE_NORMAL_VERTEX_V),
+        dual_normal_cell_x=geometry_field_source.get(geometry_meta.EDGE_TANGENT_CELL_U),
+        dual_normal_cell_y=geometry_field_source.get(geometry_meta.EDGE_TANGENT_CELL_V),
+        dual_normal_vert_x=geometry_field_source.get(geometry_meta.EDGE_TANGENT_VERTEX_U),
+        dual_normal_vert_y=geometry_field_source.get(geometry_meta.EDGE_NORMAL_VERTEX_V),
+    )
+
+    topo_c = jablonowski_williamson_topography(
+        cell_lat=cell_geometry.cell_center_lat.asnumpy(),
```

Comment:
```suggestion
        cell_lat=cell_geometry.cell_center_lat.ndarray,
```
the `.asnumpy()` does a device to host copy if you run on GPU, but you want a cupy array in this case.